libvast/src/value_index_factory.cpp

```diff
@@ -29,17 +29,19 @@ namespace {
 
 template <class T>
 size_t extract_max_size(const T& x, size_t default_value = 1024) {
-  if (auto a = extract_attribute(x, "max_size"))
-    if (auto max_size = to<size_t>(*a))
-      return *max_size;
+  if (auto a = find_attribute(x, "max_size"))
+    if (auto value = a->value)
+      if (auto max_size = to<size_t>(*value))
+        return *max_size;
   return default_value;
 }
 
 template <class T>
 optional<base> parse_base(const T& x) {
-  if (auto a = extract_attribute(x, "base")) {
-    if (auto b = to<base>(*a))
-      return *b;
+  if (auto a = find_attribute(x, "base")) {
```

Comment:
Regarding the naming of the function: I would prefer not to do any parsing / input validation this late, meaning we should validate `load_schema()` and use an assert here instead.

Then we can rename this function to `extract_base` to synchronize with `extract_max_size`, and drop the `optional` from the return type.

This should probably be done in another PR :-).

Reply:
Fair point, attributes are currently arbitrary string key-value pairs in the schema. Our parser doesn't restrict a type or format for a specific attribute, e.g., `#base`, `#base=foo`, and `#base=42` will parse just fine. Until we tighten this restriction, we have to do parsing late in the game.

Down the line, I would like to see two things:

1. Let attributes map from a string key to `data` instead of to `std::string`.
2. For each attribute we support, decide at parse time which type is supported/expected.

If that sounds reasonable to you, then we should open another story for this.

Reply:
Sounds good to both.